src/linear_algebra/Hermitian_inner_product.lean

```diff
@@ -0,0 +1,669 @@
+/-
+Copyright (c) 2018 Andreas Swerdlow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Andreas Swerdlow
+-/
+
+import analysis.normed_space.basic linear_algebra.sesquilinear_form data.complex.basic
+
+open vector_space field set complex real
+
+set_option class.instance_max_depth 32
+
+lemma ne_comm {α : Type*} {a b : α} : a ≠ b ↔ b ≠ a :=
+⟨ λ H, iff_subst (@eq_comm _ a b) H, 
+  λ H, iff_subst (@eq_comm _ b a) H⟩ 
+
+lemma im_eq_zero_iff_conj_eq (x : ℂ) : x.im = 0 ↔ conj(x) = x :=
+begin
+  dunfold conj,
+  split; intro H,
+  { rw [H, neg_zero, ←H, complex.eta] },
+
+  { rw complex.ext_iff at H,
+    have : -x.im = x.im,
+    { exact H.right },
+    rw ←add_eq_zero_iff_neg_eq at this,
+    exact add_self_eq_zero.mp this}
+end
+
+lemma ne_zero_im_zero_imp_re_ne_zero {x : ℂ} : x ≠ 0 → x.im = 0 → x.re ≠ 0 :=
+begin
+  intros H1 H2,
+  have Hx : x = ↑x.re,
+  { rw [←re_add_im x, H2, of_real_zero, 
+        zero_mul, field.add_zero, of_real_re] },
+  rw Hx at H1,
+  exact of_real_ne_zero.mp H1,
+end
+
+lemma re_of_real {x : ℂ} : x.im = 0 → ↑(x.re) = x :=
```

Comment:
I imagine this is already somewhere in the library. Also, the name should probably be `re_of_im_zero`.